src/v/cloud_storage/cache_service.cc

```diff
@@ -989,15 +1032,65 @@ ss::future<std::optional<cache_item>> cache::get(std::filesystem::path key) {
     co_return std::optional(cache_item{std::move(cache_file), data_size});
 }
 
+static constexpr uint32_t max_num_cloud_storage_cache_buckets = 1024;
```

Comment:
nit: Not for this PR, but could this be set as a bound in the config for `cloud_storage_cache_num_buckets`? 

Since the field is `std::optional` it might need a validator.